### PR TITLE
Fix typo and remove reference to version compatibility link

### LINKED ISSUE #3588: Bad URL to LS overview page
Reported [here](https://discuss.elastic.co/t/recommended-es-version-for-logstash-docs-link-broken/25143).

https://www.elastic.co/guide/en/logstash/current/getting-started-with-logstash.html
> Make sure they match based on the [Logstash version](http://www.elastic.co/overview/logstash) you’re running!

We're linking to http://www.elastic.co/overview/logstash instead of https://www.elastic.co/products/logstash (or maybe the downloads page?).

### PR DESCRIPTION
**Pull request by suyograo**
Closes #3588

### CODE REVIEW

File: docs/asciidoc/static/getting-started-with-logstash.asciidoc [hunk 1]
```diff
@@ -119,10 +119,8 @@ cd elasticsearch-{elasticsearch_version}/
 ----------------------------------
 
 NOTE: This tutorial runs Logstash {logstash_version} with Elasticsearch
-{elasticsearch_version}, although you can use it with a cluster running 1.0.0 or
-later. Each release of Logstash has a *recommended* version of Elasticsearch you
-should use. Make sure they match based on the
-http://www.elastic.co/overview/logstash[Logstash version] you're running!
+{elasticsearch_version}, although you can use it with a Elasticsearch cluster running 1.0.0 or
```
Comment (talevy) [inline docs/asciidoc/static/getting-started-with-logstash.asciidoc]: s/a/an

**Comment by suyograo** [comment]
@talevy done

**Comment by talevy** [comment]
squash commits?

LGTM

**Comment by suyograo** [comment]
squashed

**Pull request closed without merging**